src/azul/service/avro_pfb.py

```diff
@@ -0,0 +1,451 @@
+from collections import (
+    defaultdict,
+)
+import logging
+from operator import (
+    attrgetter,
+)
+from typing import (
+    Iterable,
+    MutableMapping,
+    MutableSet,
+    cast,
+)
+import uuid
+
+import attr
+import fastavro
+from fastavro.validation import (
+    ValidationError,
+)
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    config,
+)
+from azul.indexer.document import (
+    FieldTypes,
+    null_bool,
+    null_float,
+    null_int,
+    null_int_sum_sort,
+    null_str,
+    pass_thru_int,
+    pass_thru_json,
+)
+from azul.json_freeze import (
+    freeze,
+    sort_frozen,
+)
+from azul.plugins.metadata.hca.transform import (
+    pass_thru_uuid4,
+    value_and_unit,
+)
+from azul.types import (
+    JSON,
+    MutableJSON,
+)
+
+log = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True, frozen=True, kw_only=True)
+class Relation:
+    dst_id: str
+    # A more appropriate variable name would be dst_type, but we stick with name
+    # to conform to PFB spec
+    dst_name: str
+
+    @classmethod
+    def to_entity(cls, entity: 'PFBEntity'):
+        return cls(dst_id=entity.id, dst_name=entity.name)
+
+
+pfb_namespace = uuid.UUID('3372fc62-76d2-4ada-a68a-f08d4f558da5')
+
+
+@attr.s(auto_attribs=True, frozen=True, kw_only=True)
+class PFBEntity:
```

Comment:
It looks as if you're not consistent with using the `PFB` prefix in class names.